account_payment_order_return/tests/test_account_payment_order_return.py

```diff
@@ -83,42 +83,27 @@ def test_global(self):
             }
         )
         wizard.populate()
-        self.assertTrue(len(wizard.move_line_ids), 1)
-        self.receivable_line = self.invoice.line_ids.filtered(
-            lambda x: x.account_id.internal_type == "receivable"
-        )
-        # Invert the move to simulate the payment
-        self.payment_move = self.env["account.move"].create(
-            self.invoice._reverse_move_vals(default_values={})
+        self.assertEquals(len(wizard.move_line_ids), 1)
+        payment = Form(
+            self.env["account.payment"].with_context(
+                active_model="account.move", active_ids=self.invoice.ids
+            )
         )
-        self.payment_move.action_post()
-        self.payment_line = self.payment_move.line_ids.filtered(
-            lambda x: x.account_id.internal_type == "receivable"
-        )
-        # Reconcile both
-        (self.receivable_line | self.payment_line).reconcile()
-        # prev_move_lines
-        wizard.include_returned = False
+        self.payment = payment.save()
+        self.payment.post()
         wizard.populate()
-        prev_move_lines = wizard.move_line_ids
         # Create payment return
-        self.payment_return = self.env["payment.return"].create(
-            {
-                "journal_id": self.journal.id,
-                "line_ids": [
-                    (
-                        0,
-                        0,
-                        {
-                            "partner_id": self.partner.id,
-                            "move_line_ids": [(6, 0, self.payment_line.ids)],
-                            "amount": self.payment_line.credit,
-                        },
-                    )
-                ],
-            }
-        )
+        payment_return_form = Form(self.env["payment.return"])
+        payment_return_form.journal_id = self.bank_journal
+        with payment_return_form.line_ids.new() as line_form:
+            line_form.move_line_ids.clear()
+            line_form.move_line_ids.add(
+                self.payment.move_line_ids.filtered(
+                    lambda x: x.account_id.internal_type == "receivable"
+                )
+            )
+        self.payment_return = payment_return_form.save()
         self.payment_return.action_confirm()
         wizard.include_returned = False
         wizard.populate()
-        self.assertTrue((wizard.move_line_ids - prev_move_lines), 1)
+        self.assertEquals(len(wizard.move_line_ids), 0)
```

Comment:
This is not resilient, as there can be other items, so please use the same formula as previously (you can add a comment in code for avoiding next time to forget why this is done this way).

Reply:
@pedrobaeza it is true. However, the problem previously became from the fact that the wizard was aggregating data from existing journals and partners. Now that we create those in the tests, there shouldn't be such a big issue. 
However, it is still better to make it more resilient.

Reply:
Finally it's not necessary to check if there can be other items because is used journals (bank and sales) created previously and it's not possible to exists other records, right?

Reply:
OK, it's very rare that there's something that alters the behavior, although not impossible: imagine a previously installed module creates several journal items per payment on that newly created journal. Anyway, let's keep it this way now.